eclipsesource/theia
# Add Accessibility API

#### What it does

Add AccessibilityInformation interface.
Add optional accessibilityInformation to StatusBarItem and TreeItem.
Also added the type to TimelineItem, but will create a follow-up ticket for loading the information into the DOM.
Proposed API:
Add StatusBarItemOptions (containing accessibilityInformation).
Provide window.createStatusBarItem(StatusBarItemOptions).

Update createStatusBarItem handler to handle StatusBarItems.
Added aria-labels for default StatusBarItems.

Contributed on behalf of STMicroelectronics.
#### How to test

StatusBarItems:
Check one of the `StatusBarItems`, like the `Toggle Bottom Panel`, in the Accessibilty tab of the Chrome DevTools. The `aria-label` should be recognized correctly.

TreeItems:
To simplify the testing the [tree-view-sample](https://github.com/microsoft/vscode-extension-samples/tree/main/tree-view-sample) vscode-extension could be used. Just add:

```
treeItem.accessibilityInformation = {
	label: 'somValue',
	role: 'someValue'
};
```
[here](https://github.com/microsoft/vscode-extension-samples/blob/92b34c5733e2d1640ad877e9ed4605da04709124/tree-view-sample/src/jsonOutline.ts#L125). And add the extension to Theia. The jsonOutline view tree should then also display the correct label and role.
#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
